tests/upgrades/test_contentview.py

```diff
@@ -11,105 +11,209 @@
+#         save_test_data({'test_name': test_name, 'cv_name': cv.name})
+
+#     @pytest.mark.post_upgrade(depend_on=test_cv_preupgrade_scenario)
+#     def test_cv_postupgrade_scenario(self, request, target_sat, pre_upgrade_data):
+#         """After upgrade, the existing content-view(created before upgrade) should be updated.
+
+#         :id: postupgrade-a4ebbfa1-106a-4962-9c7c-082833879ae8
+
+#         :steps:
+#           1. Check yum and file repository which was added in CV before upgrade.
+#           2. Check the content view which was was created before upgrade.
+#           3. Remove yum repository from existing CV.
+#           4. Create new yum repository in existing CV.
+#           5. Publish content-view
+
+#         :expectedresults: After upgrade,
+#           1. All the repositories should be intact.
+#           2. Content view created before upgrade should be intact.
+#           3. The new repository should be added/updated to the CV.
+
+#         """
+#         pre_test_name = pre_upgrade_data.get('test_name')
+#         cv_name = pre_upgrade_data.get('cv_name')
+#         org = target_sat.api.Organization().search(query={'search': f'name="{pre_test_name}_org"'})[
+#             0
+#         ]
+#         request.addfinalizer(org.delete)
+#         product = target_sat.api.Product(organization=org.id).search(
+#             query={'search': f'name="{pre_test_name}_prod"'}
+#         )[0]
+#         request.addfinalizer(product.delete)
+#         cv = target_sat.api.ContentView(organization=org.id).search(
+#             query={'search': f'name="{cv_name}"'}
+#         )[0]
+#         yum_repo = target_sat.api.Repository(organization=org.id).search(
+#             query={'search': f'name="{pre_test_name}_yum_repo"'}
+#         )[0]
+#         request.addfinalizer(yum_repo.delete)
+#         file_repo = target_sat.api.Repository(organization=org.id).search(
+#             query={'search': f'name="{pre_test_name}_file_repo"'}
+#         )[0]
+#         request.addfinalizer(file_repo.delete)
+#         request.addfinalizer(cv.delete)
+#         cv.repository = []
+#         cv.update(['repository'])
+#         assert len(cv.read_json()['repositories']) == 0
+
+#         yum_repository2 = target_sat.api.Repository(
+#             product=product, name=f'{pre_test_name}_yum_repos2', url=settings.repos.yum_2.url
+#         ).create()
+#         yum_repository2.sync()
+#         cv.repository = [yum_repository2]
+#         cv.update(['repository'])
+#         assert cv.read_json()['repositories'][0]['name'] == yum_repository2.name
+
+#         cv.publish()
+#         assert len(cv.read_json()['versions']) == 2
+#         content_view_json = cv.read_json()['environments'][0]
+#         cv.delete_from_environment(content_view_json['id'])
+#         assert len(cv.read_json()['environments']) == 0
+
+
+@pytest.fixture
+def cv_upgrade_setup(content_upgrade_shared_satellite, upgrade_action):
```

Comment:
I think we should only create fixtures when they are reusable in multiple tests else this could be simply part of test case.

Or we can max create functions if they are common for upgrade scenarios in general.

Reply:
because both content_upgrade_shared_satellite and upgrade_action are fixtures, this needs to be a fixture to use them